boa_engine/src/context/intrinsics.rs

```diff
@@ -0,0 +1,368 @@
+use crate::{
+    builtins::{error::r#type::create_throw_type_error, iterable::IteratorPrototypes},
+    object::{JsObject, ObjectData},
+    Context,
+};
+
+#[derive(Debug, Default)]
+pub struct Intrinsics {
```

Comment:
It's forced, since the `intrinsics` method of `Context` is public and we cannot return a non-public type from a public method.

Reply:
True, I was more thinking if we want to expose the `Context::intrinsics` method at all. Currently we expose the `Context::standard_objects`, but is that actually useful outside of the builtins? But it's probably best to discuss this together with other public API topics.

Reply:
I think exposing it is necessary since this is the only way our users could inherit from builtin prototypes with ease.

What we could do is hide every other intrinsic and just leave the constructors public.